rootfs/scheduler/__init__.py

```diff
@@ -1094,6 +1098,42 @@ def _buildpack_readiness_probe(self, controller, delay=30, timeout=5, period_sec
 
         return controller
 
+    '''
+    Applies tcp socket readiness probe to the docker app container only if some port is exposed
+    by the docker image.
+    '''
+    def _dockerapp_readiness_probe(self, controller, image, delay=5, timeout=5, period_seconds=5,
+                                   success_threshold=1, failure_threshold=1):
+        port = self._get_port(image)
+        if port is None:
+            return controller
+        readinessprobe = {
```

Comment:
Shipping with >10 lines of duplicate code (especially one that mimics the other, save for the probe) is never a good idea, refactoring or not. There's no technical reason why we shouldn't refactor this so there's less duplicate code other than time.